### PR TITLE
fix: Detect replacement tx when creating a safe

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Calling `provider.waitForTransaction` with the cached `txHash` will not detect if the transaction was cancelled or sped-up. Instead we now call `provider._waitForTransaction` with a `replacement` object and handle those cases.

## How to test

1. Create a new safe
2. Reload the page while the tx is executing
3. Cancel or speed-up the transaction in your wallet
4. Observe the Status changing to ERROR or SUCCESS

### CODE REVIEW

File: src/components/create-safe/status/usePendingSafeCreation.ts [hunk 1]
```diff
@@ -19,18 +24,42 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
...
     if (didRevert(receipt)) {
       return SafeCreationStatus.REVERTED
     }
 
     return SafeCreationStatus.SUCCESS
-  } catch (error) {
+  } catch (err) {
+    const error = err as EthersError
+
+    if (error.code === 'TRANSACTION_REPLACED') {
```
Comment (iamacook) [inline src/components/create-safe/status/usePendingSafeCreation.ts:54]:
```suggestion
    if (error.code === ErrorCode.TRANSACTION_REPLACED) {
```

File: src/components/create-safe/status/usePendingSafeCreation.ts [hunk 2]
```diff
@@ -19,18 +24,42 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
...
+      if (error.reason === 'cancelled') {
+        return SafeCreationStatus.ERROR
+      } else {
+        console.log('speed-up tx found!')
```
Comment (iamacook) [inline src/components/create-safe/status/usePendingSafeCreation.ts:58]: If we intend to log this, can we provide more information, i.e. the transaction hash?

File: src/components/create-safe/status/usePendingSafeCreation.ts [hunk 3]
```diff
@@ -19,18 +24,42 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
...
+    const txResponse = await provider.getTransaction(txHash)
+    const proxyContractAddress = getProxyFactoryContractInstance(chainId).address
+
+    const replacement = {
+      data: txResponse.data,
+      from: txResponse.from,
+      nonce: txResponse.nonce,
+      to: proxyContractAddress,
+      value: txResponse.value,
+      startBlock: txResponse.blockNumber || blockNumber,
```
Comment (iamacook) [inline src/components/create-safe/status/usePendingSafeCreation.ts:41]: If a transaction already exists, we are still requesting the current block number as well.

```suggestion
    const txResponse = await provider.getTransaction(txHash)
    const proxyContractAddress = getProxyFactoryContractInstance(chainId).address

    const replacement = {
      data: txResponse.data,
      from: txResponse.from,
      nonce: txResponse.nonce,
      to: proxyContractAddress,
      value: txResponse.value,
      startBlock: txResponse.blockNumber || await provider.getBlockNumber(),
```

File: src/components/create-safe/status/usePendingSafeCreation.ts (hunk 2 above)
Comment (usame-algan) [reply src/components/create-safe/status/usePendingSafeCreation.ts:58]: Ah, good catch! It was a leftover. I will remove it

File: src/components/create-safe/status/usePendingSafeCreation.ts (hunk 3 above)
Comment (usame-algan) [reply src/components/create-safe/status/usePendingSafeCreation.ts:41]: Nice!

**Review by iamacook — APPROVED** [review]

**Review by katspaugh — APPROVED** [review]
🚀

**Pull request merged**